execution/binary/scalar.go

```diff
@@ -122,6 +123,8 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 		o.next.GetPool().PutStepVector(vector)
 	}
 	o.next.GetPool().PutVectors(in)
+	o.scalar.GetPool().PutVectors(scalarIn)
```

Comment:
I think we also need to call `PutStepVector` for the scalar operator at one point.

Reply:
Ack will do!